### PR TITLE
Deprecate `Trainer.should_rank_save_checkpoint` property

### PR DESCRIPTION
**Pull request by jjenniferdai**
## What does this PR do?

see context: https://github.com/PyTorchLightning/pytorch-lightning/pull/11035#discussion_r767079698
original PR: #9433
original issue: #9074

Go through deprecation for public property

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ananthsub — APPROVED** [review]

File: pytorch_lightning/trainer/trainer.py [hunk 1]
```diff
@@ -1699,6 +1699,11 @@ def world_size(self) -> int:
         # some training types define a world size
         return getattr(self.training_type_plugin, "world_size", 1)
 
+    @property
+    def should_rank_save_checkpoint(self) -> bool:
+        rank_zero_deprecation("`Trainer.should_rank_save_checkpoint` is deprecated in v1.6 and will be removed in 1.8.")
```
Comment (carmocca) [inline pytorch_lightning/trainer/trainer.py:1704]:
```suggestion
        rank_zero_deprecation("`Trainer.should_rank_save_checkpoint` is deprecated in v1.6 and will be removed in 1.8.", stacklevel=5)
```

Properties should increase the stacklevel by 1.

File: CHANGELOG.md [hunk 2]
```diff
@@ -604,7 +607,6 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Removed `TrainerProperties` mixin and moved property definitions directly into `Trainer` ([#9495](https://github.com/PyTorchLightning/pytorch-lightning/pull/9495))
 - Removed a redundant warning with `ModelCheckpoint(monitor=None)` callback ([#9875](https://github.com/PyTorchLightning/pytorch-lightning/pull/9875))
 - Remove `epoch` from `trainer.logged_metrics` ([#9904](https://github.com/PyTorchLightning/pytorch-lightning/pull/9904))
-- Removed `should_rank_save_checkpoint` property from Trainer ([#9433](https://github.com/PyTorchLightning/pytorch-lightning/pull/9433))
```
Comment (carmocca) [inline CHANGELOG.md:607]: Oh, I didn't notice this had been already included in the 1.5 release.
In that case, it's too late to deprecate to avoid breaking BC.

@ananthsub do you still think we should add this back? If yes, then this should be included in a bug-fix release

Comment (ananthsub) [reply CHANGELOG.md:607]: Oh good catch @carmocca - I didn't realize it was removed already in the public release either. In that case, I don't think we should add it back as the breaking change is already done :/

Comment (jjenniferdai) [reply CHANGELOG.md:607]: changelog still has this "removal" listed but it wasn't actually removed bc of the merging mistake? (described in #11035)

**Review by ananthsub — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]
I don't understand how we can deprecate this on the trainer while at the same time remove it on the strategy. You will have to copy this implementation into the deprecated property method:

`return isinstance(self.ttp, TPUSpawn) and self.local_rank == 0 or self.is_global_zero ` and it won't cover custom plugins.

So yeah, this will be confusing.
